viadot/sources/velux_club.py

```diff
@@ -0,0 +1,186 @@
+# Customed Errors!
+class Source_NOK(Exception):
+    def __init__(self, message):
+        self.message = message
+    def __str__(self):
+        return self.message
+    
+class Dates_NOK(Exception):
+    def __init__(self, message):
+        self.message = message
+    def __str__(self):
+        return self.message
+
+class Historical_Too_Old(Exception):
+    def __init__(self, message):
+        self.message = message
+    def __str__(self):
+        return self.message
+    
+class VeluxClub(Source):
+
+    """
+    A class implementing the Velux Club API.
+
+    Documentation for this API is located at: https://api.club.velux.com/api/v1/datalake/
+    There are 4 endpoints where to get the data!
+
+    Parameters
+    ----------
+    query_params : Dict[str, Any], optional
+        The parameters to pass to the GET query.
+        See https://supermetrics.com/docs/product-api-get-data/ for full specification,
+        by default None
+    """
+
+    API_URL = "https://api.club.velux.com/api/v1/datalake/"
+
+    def __init__(
+        self,
+        *args,
+        credentials: Dict[str, Any] = None,
+        **kwargs,
+    ):
+        DEFAULT_CREDENTIALS = get_source_credentials("velux_club")
+        credentials = kwargs.pop("credentials", DEFAULT_CREDENTIALS)
+        if credentials is None:
+            raise CredentialError("Missing credentials.")
+
+        self.headers = {
+            "Authorization": "Bearer " + credentials["token"],
+            "Content-Type": "application/json",
+        }
+
+        super().__init__(*args, credentials=credentials, **kwargs)
+
+    def build_query(
+        self, source: str, from_date: str, to_date: str, api_url: str, region: str
+    ) -> str:
+        if source in ["jobs", "product", "company"]:
+            # check if date filter was passed!
+            if from_date == "" or to_date == "":
+                return "Introduce a 'FROM Date' and 'To Date'"
+            url = f"{api_url}{source}?from={from_date}&to={to_date}&region&limit=100"
+        elif source in "survey":
+            url = f"{api_url}{source}?language=en&type=question"
+        else:
+            return "pick one these sources: jobs, product, company, survey"
+        return url
+
+    def get_response(
+        self,
+        source: str = "",
+        from_date: str = "2022-03-22",
+        to_date: str = "",
+        region="null",
+    ) -> pd.DataFrame:  ## Returns the response
+        # DocString
+
+        # Dealing with bad arguments
+        if (source not in ["jobs", "product", "company", "survey"]):
+            raise Source_NOK("The source has to be: jobs, product, company or survey")
+
+        from_date_obj = datetime.strptime(from_date, '%Y-%m-%d')  
+        oldest_date_obj = datetime.strptime("2022-03-22", '%Y-%m-%d')           
+        delta = from_date_obj - oldest_date_obj  
+        
+        if (delta.days <0 ):
+            raise Historical_Too_Old("from_date has to be earlier than 2023-03-22!!")
```

Comment:
Why? This looks like an arbitrary number. Pls explain in a comment why this specific value was chosen.